crates/outbound-redis/src/lib.rs

```diff
@@ -2,13 +2,29 @@ use outbound_redis::*;
 use redis::Commands;
 
 pub use outbound_redis::add_to_linker;
+use spin_engine::host_component::HostComponent;
 
 wit_bindgen_wasmtime::export!("../../wit/ephemeral/outbound-redis.wit");
 
 /// A simple implementation to support outbound Redis commands.
 #[derive(Default, Clone)]
 pub struct OutboundRedis;
 
+impl HostComponent for OutboundRedis {
+    type Data = Self;
```

Comment:
I don't quite get why this would be `Self` rather than `()`.  Is the 'runtime state' always an instance of the component plus optional something else (config?)?  If so, would it be clearer to distinguish these, so that the Self is baked in and only the additional data needs to be in the `Data` type?

Reply:
The closure passed to `add_to_linker` just below has to return a `&mut Data`; you have to have an instance of `Data` to borrow from even if its a unit struct (unless you want to do some `transmute` jiggery-pokery, but if your `Data` has state you'll need the ref anyway).

`Data` is just the type that implements the interface _unless_ that interface uses resources, in which case `Data` is `(Impl, <InterfaceName>Tables<Impl>)` (the "tables" track resource state).

I would be happy to have a way to simplify this, but I spent way too much time trying and not enough time succeeding.

Reply:
To clarify: "has to return a `&mut Data`" really means "`Data` has to match the wit-bindgen-generated signature of that closure".